### ISSUE TITLE
EC2 network scripts are broken

### ISSUE THREAD

**Issue by chipzz** [andsens/bootstrap-vz#324]
The scripts in https://github.com/andsens/bootstrap-vz/tree/master/bootstrapvz/providers/ec2/assets/ec2 are incredibly broken; they're meant for a RedHat system and do not work and are downright harmful in a Debian environment; they will also run on random interfaces (like tunnel interfaces) and wreak havoc to the point were it will tear down the interface again and you have to reboot to fix it (due to a missing dependency on curl). Also they're not properly installed, /etc/sysconfig is a RedHat-ism that does not work on Debian, at all.

I've done some heavy work on packaging these scripts, fixing a whole slew of issues, like packaging them in a Debian package, adding a wget dependency, putting the scripts in the correct place outside of /etc, removing all bashisms so the DHCP script which is sourced can actually run without blowing up if /bin/sh -/-> /bin/bash, and fixing the udev rules so they only run on eth* interfaces.
The end result of this being that now the scripts actually work as intended (or at least, part of them; another part of them appears to be buggy the way Amazon shipped them for RedHat in the first place).

Please replace the files in your repo with copies of the files in my repo (and adjust paths accordingly).

**Comment by andsens** [comment]
Well f*** :-/
I had no clue and assumed that if James shipped them with the official images they'd be *good enough*.
I'll review them and replace the current files.

Thank you very much for your work @chipzz.

@JamesBromberger, could you maybe sign off on this as well?

**Comment by nbraud** [comment]
@chipzz Could you, perchance, consider uploading `ec2net` to the Debian archive?
It is broadly useful (for EC2 users), and having it managed through apt means that it is possible to get updates (and I think it is cleaner for bootstrap-vz to install a package than to blast files in the image).

**Comment by andsens** [comment]
Agreed. I would much prefer that approach, this also means less code to maintain.

**Comment by chipzz** [comment]
I have considered that myself and agree it would be the preferable approach.
Unfortunately I'm not a DD, so I can't upload this myself, and I have asked around a couple of times in #debian-cloud already, with zero response, so I don't think there's currently a lot of interest to have this package in Debian unfortunately. :(

I think we have 4 options:
a) status quo, no changes at all;
b) remove the current scripts from your repo;
c) import of my version of the scripts in your repo;
d) get the package in Debian and install it that way.

It looks like d) is currently not an option. I don't think a) would be a good solution, I don't mean to be rude but these files in their current form cause more harm than good. If you're not prepared to import my version, I would suggest you at least remove the current version.

I would obviously prefer solution c). Is there anything I can do to convince you to go that route, or assist in any way? Please note however that I currently do not have bootstrap-vz set up myself.

**Comment by nbraud** [comment]
@chipzz You don't need to be a DD for option d: it's possible to upload the package to [mentors](https://mentors.debian.net) and [request a sponsored upload](https://wiki.debian.org/DebianMentorsFaq#How_do_I_get_a_sponsor_for_my_package.3F).

(c) could be a stop-gap, but we can get the upload done fairly quick I would guess.
